ext/distributed_tracing_headers.c

```diff
@@ -0,0 +1,400 @@
+                        }
+                        char *valueend = ptr;
+                        while (*valueend == ' ' || *valueend == '\t') {
+                            --valueend;
+                        }
+                        size_t valuelen = valueend - valuestart;
+
+                        if (keylen == 1 && keystart[0] == 's') {
+                            int extraced_priority = strtol(valuestart, NULL, 10);
+                            if ((result.priority_sampling > 0) == (extraced_priority > 0)) {
+                                result.priority_sampling = extraced_priority;
+                            } else {
+                                result.conflicting_sampling_priority = true;
+                            }
+                        } else if (keylen == 1 && keystart[0] == 'o') {
+                            if (result.origin) {
+                                zend_string_release(result.origin);
+                            }
+                            result.origin = zend_string_init(valuestart, valuelen, 0);
+                            for (char *valptr = ZSTR_VAL(result.origin), *valend = valptr + valuelen; valptr < valend; ++valptr) {
+                                if (*valptr == '~') {
+                                    *valptr = '=';
+                                }
+                            }
+                        } else if (keylen > 2 && keystart[0] == 't' && keystart[1] == '.') {
+                            zend_string *tag_name = zend_strpprintf(0, "_dd.p.%.*s", (int) keylen - 2, keystart + 2);
+                            zval zv;
+                            ZVAL_STRINGL(&zv, valuestart, valuelen);
+                            for (char *valptr = Z_STRVAL(zv), *valend = valptr + valuelen; valptr < valend; ++valptr) {
+                                if (*valptr == '~') {
+                                    *valptr = '=';
+                                }
+                            }
+                            zend_hash_update(&result.meta_tags, tag_name, &zv);
+                            zend_hash_add_empty_element(&result.propagated_tags, tag_name);
+                            zend_string_release(tag_name);
+                        } else {
+                            zval zv;
+                            ZVAL_STRINGL(&zv, valuestart, valuelen);
+                            zend_hash_str_update(&result.tracestate_unknown_dd_keys, keystart, keylen, &zv);
+                        }
+                    } while (*ptr == ';');
+
+                    continue;
+                }
+                *(persist++) = *ptr;
+
+                if (*ptr == ' ' || *ptr == '\t') {
+                    continue;
+                }
+
+                last_comma = *ptr == ',';
+                // preserve only up to 31 vendor specific values, excluding our own
+                if (last_comma && ++commas == 30) {
+                    --persist;
+                    break;
+                }
```

Comment:
I know this is not a change associated with this PR, nor is it a blocker, but would you happen to know why this is the case?

Reply:
Because w3c header consumers are allowed to discard any tracestate exceeding 32 vendor specific values.